sklearn/gaussian_process/gpr.py

```diff
@@ -241,9 +241,14 @@ def obj_func(theta, eval_gradient=True):
         # of actual query points
         K = self.kernel_(self.X_train_)
         K[np.diag_indices_from(K)] += self.alpha
-        self.L_ = cholesky(K, lower=True)  # Line 2
-        self.alpha_ = cho_solve((self.L_, True), self.y_train_)  # Line 3
+        try:
+            self.L_ = cholesky(K, lower=True)  # Line 2
+        except np.linalg.LinAlgError:
+            raise ValueError("The DotProduct kernel isn't returning a "
```

Comment:
The problem is not necessarily about the DotProductKernel it could potentially happen with any kind of kernel. Use "The kernel", maybe put the kernel used (self.kernel_) as part of the error message.

Another potential improvement in the message:
"Try gradually increasing the 'alpha' parameter of your GaussianProcessRegressor estimator"